packages/js/src/ui/components/primitives/Tabs/TabsTab.tsx

```diff
@@ -0,0 +1,42 @@
+import { JSX, ParentProps, Ref } from 'solid-js';
+import { cn } from '../../../helpers';
+import { AppearanceKey } from '../../../context';
+import { Button } from '../Button';
+import { useTabsContext } from './Tabs';
+
+type TabsTabProps = ParentProps & {
+  value: string;
+  class?: string;
+  appearanceKey?: AppearanceKey;
+  ref?: Ref<HTMLButtonElement>;
+  onClick?: JSX.EventHandlerUnion<HTMLButtonElement, MouseEvent>;
+};
+
+export const TabsTab = (props: TabsTabProps) => {
+  const { activeTab, setActiveTab } = useTabsContext();
+  const clickHandler = () => setActiveTab(props.value);
+
+  return (
+    <Button
+      variant="unstyled"
+      size="none"
+      ref={props.ref}
+      id={props.value}
+      appearanceKey={props.appearanceKey ?? 'tabsTab'}
+      class={cn(
+        `nt-relative nt-transition nt-outline-none focus-visible:nt-outline-none focus-visible:nt-ring-2 focus-visible:nt-ring-primary focus-visible:nt-ring-offset-2 nt-pb-[0.625rem] after:nt-absolute after:nt-content-[''] after:nt-bottom-0 after:nt-left-0 after:nt-w-full after:nt-h-[2px] after:nt-border-b-2`,
+        activeTab() === props.value
```

Comment:
Instead of using a signal here, let's just add a data attribute (`data-selected`) when this is the active tab and use the selector to style! This will also enable developer to target just the active tab via CSS.

Reply:
I think we need to be using data attributes, as that's what's most commonly used in these cases:
Take this as an example: https://www.radix-ui.com/primitives/docs/components/tabs#trigger

Reply:
yeah for the better DX, we can support both ;)